os/arch/arm/src/amebad/amebad_i2c.c

```diff
@@ -0,0 +1,1241 @@
+	//.registercallback = amebad_i2c_registercallback
+#endif
+};
+
+/* I2C device structures */
+
+static const struct amebad_i2c_config_s amebad_i2c1_config = {
+	//.base = AMEBAD_I2C1_BASE,
+	//.busy_idle = CONFIG_I2C1_BUSYIDLE,
+	//.filtscl = CONFIG_I2C1_FILTSCL,
+	//.filtsda = CONFIG_I2C1_FILTSDA,
+	.scl_pin = PA_25,
+	.sda_pin = PA_26,
+#ifndef CONFIG_I2C_SLAVE
+	.mode = AMEBAD_I2C_MASTER,
+#else
+	.mode = AMEBAD_I2C_SLAVE,
+#endif
+#ifndef CONFIG_I2C_POLLED
+	.irq = AMEBAD_IRQ_I2C0,
+#endif
+};
+
+static struct amebad_i2c_priv_s amebad_i2c1_priv = {
+	.ops = &amebad_i2c_ops,
+	.config = &amebad_i2c1_config,
+	.refs = 0,
+	.intstate = INTSTATE_IDLE,
+	.msgc = 0,
+	.msgv = NULL,
+	.ptr = NULL,
+	.dcnt = 0,
+	.flags = 0,
+	.status = 0
+};
+
+/************************************************************************************
+ * Private Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Name: amebad_i2c_getreg
+ *
+ * Description:
+ *   Get a 32-bit register value by offset
+ *
+ ************************************************************************************/
+
+static inline uint32_t amebad_i2c_getreg(FAR struct amebad_i2c_priv_s *priv, uint16_t offset)
+{
+	//return getreg32(priv->config->base + offset);
+	return 0;
+}
+
+/************************************************************************************
+ * Name: amebad_i2c_putreg
+ *
+ * Description:
+ *  Put a 32-bit register value by offset
+ *
+ ************************************************************************************/
+
+static inline void amebad_i2c_putreg(FAR struct amebad_i2c_priv_s *priv, uint16_t offset, uint32_t value)
+{
+	//putreg32(value, priv->config->base + offset);
+}
+
+/************************************************************************************
+ * Name: amebad_i2c_modifyreg
+ *
+ * Description:
+ *   Modify a 32-bit register value by offset
+ *
+ ************************************************************************************/
+
+static inline void amebad_i2c_modifyreg(FAR struct amebad_i2c_priv_s *priv, uint16_t offset, uint32_t clearbits, uint32_t setbits)
+{
+	//modifyreg32(priv->config->base + offset, clearbits, setbits);
+}
+
+/************************************************************************************
+ * Name: amebad_i2c_sem_wait
+ *
+ * Description:
+ *   Take the exclusive access, waiting as necessary
+ *
+ ************************************************************************************/
+
+static inline void amebad_i2c_sem_wait(FAR struct amebad_i2c_priv_s *priv)
+{
+	int ret;
+
+	do {
+		/* Take the semaphore (perhaps waiting) */
+
+		ret = sem_wait(&priv->sem_excl);
+
+		/* The only case that an error should occur here is if the wait was
+		 * awakened by a signal.
+		 */
+
+		DEBUGASSERT(ret == OK || ret == -EINTR);
```

Comment:
`ret` cannot be `-EINTR`. If you want to check `EINTR`, you can use errno instead of return.